File: Algorithms/python/3151-special_array_I.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        
        i = 0
        while i + 1 < len(nums):
            if nums[i] % 2 == 0 and nums[i + 1] % 2 == 0:
                return False
            elif nums[i] % 2 != 0 and nums[i + 1] % 2 != 0:
                return False
            i += 1
        return True

File: Algorithms/C++/3151_special_array_I.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        
        if(nums.size() == 1){
            return true;
        }

        for(int i = 0; i + 1 < nums.size(); i++){
            
            if(even(nums[i]) == even(nums[i + 1])){
                return false;
            }
        }

        return true;
    }

    bool even(int n){
        return n % 2 == 0;
    }
};